# **Libraries**

In [7]:
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np
import seaborn as sns

# **Data Loading**

In [8]:
df_train = pd.read_csv("train_features.csv")

In [9]:
df_train.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3817 entries, 0 to 3816
Data columns (total 16 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   tahun_kelahiran          3817 non-null   int64  
 1   pendidikan               3628 non-null   object 
 2   status_pernikahan        3605 non-null   object 
 3   pendapatan               3627 non-null   float64
 4   jumlah_anak_balita       3627 non-null   float64
 5   jumlah_anak_remaja       3613 non-null   float64
 6   terakhir_belanja         3645 non-null   float64
 7   belanja_buah             3636 non-null   float64
 8   belanja_daging           3639 non-null   float64
 9   belanja_ikan             3624 non-null   float64
 10  belanja_kue              3603 non-null   float64
 11  pembelian_diskon         3639 non-null   float64
 12  pembelian_web            3652 non-null   float64
 13  pembelian_toko           3648 non-null   float64
 14  keluhan                 

**insight:**
1. Total baris keseluruhan ada 3817
2. Yang tidak missing value hanya kolom tahun_kelahirn saja, selain itu missing value
3. Type data kolom tangga_menjadi_anggota harusnya datetime

In [10]:
def is_decimal(df, columns):
    for col in columns:
        decimals_exist = False
        for value in df[col].dropna():
            if value % 1 != 0:
                decimals_exist = True
                break
        if decimals_exist:
            print("decimal exist")
        else:
            print("decimal not exit")

# Contoh penggunaan
cols = [ 'pendapatan', 'jumlah_anak_balita', 'jumlah_anak_remaja', 'terakhir_belanja', 'belanja_buah', 'belanja_daging', 'belanja_ikan', 'belanja_kue', 'pembelian_diskon', 'pembelian_web', 'pembelian_toko', 'keluhan']
is_decimal(df_train, cols)

decimal not exit
decimal not exit
decimal not exit
decimal not exit
decimal not exit
decimal not exit
decimal not exit
decimal not exit
decimal not exit
decimal not exit
decimal not exit
decimal not exit


**insight :**
Tujuan pengecekan nilai desimal pada fitur bertipe float adalah untuk memastikan data konsisten dan valid.Beberapa fitur, seperti jumlah_anak_balita tidak mungkin memiliki nilai decimal

In [11]:
df_train.describe()

tahun_kelahiran    pendapatan  jumlah_anak_balita  jumlah_anak_remaja  \
count      3817.000000  3.627000e+03         3627.000000         3613.000000   
mean       1967.823946  1.144832e+08            0.293080            0.353723   
std          11.768131  4.346042e+07            0.473063            0.493014   
min        1899.000000  5.073000e+06            0.000000            0.000000   
25%        1959.000000  8.112512e+07            0.000000            0.000000   
50%        1968.000000  1.156214e+08            0.000000            0.000000   
75%        1976.000000  1.504960e+08            1.000000            1.000000   
max        2000.000000  3.328840e+08            2.000000            2.000000   

       terakhir_belanja   belanja_buah  belanja_daging   belanja_ikan  \
count       3645.000000    3636.000000    3.639000e+03    3624.000000   
mean          47.231550   59804.239824    4.385748e+05   81428.997792   
std           27.068512   74024.976109    5.120427e+05   99976.226855   
min            0.000000       0.000000    0.000000e+00       0.000000   
25%           25.000000    7907.000000    4.947950e+04   10115.000000   
50%           47.000000   26456.000000    2.219930e+05   36054.500000   
75%           69.000000   86162.000000    6.863555e+05  121380.000000   
max          128.000000  396508.000000    3.489675e+06  621600.000000   

         belanja_kue  pembelian_diskon  pembelian_web  pembelian_toko  \
count    3603.000000       3639.000000    3652.000000     3648.000000   
mean    63377.970580          2.125584       4.436473        5.767818   
std     79435.457282          2.100133       3.002522        3.210738   
min         0.000000          0.000000       0.000000        0.000000   
25%      7947.000000          0.000000       2.000000        3.000000   
50%     27795.000000          2.000000       4.000000        5.000000   
75%     89502.500000          3.000000       7.000000        8.000000   
max    542164.000000         20.000000      30.000000       17.000000   

           keluhan  
count  3621.000000  
mean      0.004971  
std       0.070340  
min       0.000000  
25%       0.000000  
50%       0.000000  
75%       0.000000  
max       1.000000

**insight:**
1. dilihat dari jarak antara mean dan median serta quartil,beberapa fitur kemungkinan memiliki distribusi normal
2. semua fitur belanja memiliki distribusi skew
3. dilihat dari jarak antar quartil dan nilai max,selain fitur jumlah anak memiliki outlier
4. varians dari tiap fitur bernilai positif
5. beberapa fitur memiliki nilai nya tinggi tinggi,maka di perlukan scalling data


In [19]:
df_train.duplicated().sum()

0

**insight:**
tidak ada data duplicate

In [20]:
y_train = pd.read_csv('train_labels.csv')

In [21]:
y_train

jumlah_promosi
0                  2
1                  0
2                  1
3                  4
4                  4
...              ...
3812               5
3813               1
3814               0
3815               0
3816               0

[3817 rows x 1 columns]

# **EDA**

1. Bagaimana missing value terjadi?
2. Bagaimana distribusi data di tiap fitur?
3. Apakah terdapat outlier?
4. Apakah datanya balance?